docs/coordinates/representations.rst

```diff
@@ -314,6 +314,138 @@ provides the ``base``::
   <SphericalRepresentation (lon, lat, distance) in (rad, rad, kpc)
       ( 0.0048481,  1.04719246,  1.00000294)>
 
+.. Note:: At present, the differential classes are generally meant to work with
+   first derivatives, but they do not check the units of the inputs to enforce
+   this. Passing in 2nd derivatives, e.g., acceleration values (with
+   acceleration units), will succeed, but any transformations that occur through
+   re-representation of the differential will not necessarily be correct.
```

Comment:
Duh, I clearly was too tired yesterday - yes, of course. And, no, the machinery does not (yet) allow that -- I think it will need an `order` keyword or something like that for `scale_factors` - makes me thing that we'd need something information on derivatives - and should get at least that part of the `Variable` work in: https://github.com/astropy/astropy/pull/3715/files#diff-2a3560744629608434d0eff93c50cf2aR13

This also starts me liking more and more @StuartLittlefair's idea of using the unit of the derivative as the key to a `dict` of differentials - that would seem to extend nicely to possible other uses.